gensim/matutils.py

```diff
@@ -1127,7 +1127,7 @@ def qr_destroy(la):
     qr, tau, work, info = geqrf(a, lwork=work[0], overwrite_a=True)
     del a  # free up mem
     assert info >= 0
-    r = triu(qr[:n, :n])
+    r = np.triu(qr[:n, :n])
```

Comment:
Does this work for all versions of scipy and numpy that Gensim supports?

Reply:
yes I see the numpy has this method added way long back in 1.13 and now the stable version is 1.26, for the existing users who are using scipy as 1.12.0 and below won't have any problem, but for users using latest scipy would need this fix

Reply:
It's probably best to conditionally import triu from np, falling back to scipy if np.triu is unavailable.

Reply:
so try and except for this if try block np.triu fails then in except import the triu from scipy and the execute the triu function ?

Reply:
At module scope:

```python
try:
    from numpy import triu
except ImportError:
    from scipy.linalg import triu
```

Reply:
Ohh I see, Thanks let me update the PR :)

Reply:
@Luffy610 Did a quick look at when NumPy added this functionality. It seems to have been in roughly 2004 or 2005.

https://github.com/numpy/numpy/commit/2d0b625d58a48525ded47c4d184ad3837e71c316#diff-fc980d30a327648402874bd5daed1e37c96153a4a8fa542d4471e0f45dfadf9fR103

(The reason why all of the files have SciPy in the name is that this release pre-dates the split between SciPy and NumPy in 2006. That's also why SciPy has a function that does the exact same thing as NumPy.)

For context, the first NumPy release which was compatible with Python 3 was 1.5.0, in 2010. So if Gensim requires that its users have Python 3, then they definitely have `np.triu`.